server/src/lib.rs

```diff
@@ -127,6 +127,8 @@ impl Server {
                     || !spawns.nodes.is_empty()
                     || !spawns.block_updates.is_empty()
                     || !spawns.voxel_data.is_empty()
+                    || !spawns.inventory_additions.is_empty()
+                    || !spawns.inventory_removals.is_empty()
```

Comment:
Another option could be to move this check to the creation of the `Spawns` struct. If `server::sim::step` returns an `Option<Spawns>` instead of a `Spawns`, we could effectively move these checks all the way to the function that creates `Spawns` to begin with. I'll try that out.